runtime/compiler/x/codegen/AllocPrefetchSnippet.cpp

```diff
@@ -232,8 +232,9 @@ uint8_t* TR::X86AllocPrefetchSnippet::emitSharedBody(uint8_t* prefetchSnippetBuf
    //
    for (int32_t lineOffset = 0; lineOffset < numLines; ++lineOffset)
       {
+      TR_ASSERT(TR::CodeGenerator::getX86ProcessorInfo().isAMD15h() == TR::comp()->target().cpu.is(OMR_PROCESSOR_X86_AMDFAMILY15H), "OMR_PROCESSOR_X86_AMDFAMILY15H\n");
       prefetchSnippetBuffer[0] = 0x0F;
-      if (cpuInfo.isAMD15h())
+      if (TR::comp()->target().cpu.is(OMR_PROCESSOR_X86_AMDFAMILY15H))
```

Comment:
Do we have comp() object somewhere handy so that we can avoid the TLS lookup?

Reply:
Its caller has a `cg` pointer that can be passed to this function. I'll make that change.

```cpp
void TR::createCCPreLoadedCode(uint8_t *CCPreLoadedCodeBase, uint8_t *CCPreLoadedCodeTop, void ** CCPreLoadedCodeTable, TR::CodeGenerator *cg)
   {
   uint8_t *cursor = CCPreLoadedCodeBase;

   CCPreLoadedCodeTable[TR_CCPreLoadedCode::TR_AllocPrefetch] = static_cast<void *>(cursor);
   if (cg->comp()->target().is64Bit())
      cursor = TR::X86AllocPrefetchSnippet::emitSharedBody<TR::HeapTypes::ZeroedHeap, true>(cursor);
   else
      cursor = TR::X86AllocPrefetchSnippet::emitSharedBody<TR::HeapTypes::ZeroedHeap, false>(cursor);

   cursor = static_cast<uint8_t *>( TR::alignAllocation<32>(cursor) );

   CCPreLoadedCodeTable[TR_CCPreLoadedCode::TR_NonZeroAllocPrefetch] = static_cast<void *>(cursor);
   if (cg->comp()->target().is64Bit())
      cursor = TR::X86AllocPrefetchSnippet::emitSharedBody<TR::HeapTypes::NonZeroedHeap, true>(cursor);
   else
      cursor = TR::X86AllocPrefetchSnippet::emitSharedBody<TR::HeapTypes::NonZeroedHeap, false>(cursor);

   cursor = static_cast<uint8_t *>( TR::alignAllocation<32>(cursor) );

   TR_ASSERT(cursor == CCPreLoadedCodeTop, "The expected and actual sizes of the emitted code differ. cursor = %p, CCPreLoadedCodeTop = %p", cursor, CCPreLoadedCodeTop);
   }
```